src/Workspaces/Remote/Core/Services/SolutionCreator.cs

```diff
@@ -497,5 +507,47 @@ private Project AddDocument(Project project, DocumentInfo documentInfo)
         {
             return project.Solution.AddDocument(documentInfo).GetProject(project.Id);
         }
+
+        private CompilationOptions FixUpCompilationOptions(ProjectInfo.ProjectAttributes info, CompilationOptions compilationOptions)
+        {
+            // see AbstractProject.CreateCompilationOptions for more detail
+            var projectDirectory = GetContainingDirectoryPathOpt(info.FilePath);
```

Comment:
Do we need to do a null or empty check on this?  (or can only C#/VB projects get here?)  I know TypeScript has a Project with an empty FilePath.